### ISSUE TITLE
radeon vii and ad 19.2.2 driver wont start

### ISSUE THREAD

**Issue by lesjokolat** [ethereum-mining/ethminer#1845]
windows 10 pro

amd driver 19.2.2

radeon vii, vega, 64 fury x gpu installed

No OpenCL platforms found
Error: No usable mining devices found

here is cli info on radeon vii

 Device Type:                                   CL_DEVICE_TYPE_GPU
  Vendor ID:                                     1002h
  Board name:                                    AMD Radeon VII
  Device Topology:                               PCI[ B#3, D#0, F#0 ]
  Max compute units:                             60
  Max work items dimensions:                     3
    Max work items[0]:                           1024
    Max work items[1]:                           1024
    Max work items[2]:                           1024
  Max work group size:                           256
  Preferred vector width char:                   4
  Preferred vector width short:                  2
  Preferred vector width int:                    1
  Preferred vector width long:                   1
  Preferred vector width float:                  1
  Preferred vector width double:                 1
  Native vector width char:                      4
  Native vector width short:                     2
  Native vector width int:                       1
  Native vector width long:                      1
  Native vector width float:                     1
  Native vector width double:                    1
  Max clock frequency:                           1802Mhz
  Address bits:                                  64
  Max memory allocation:                         4244635648
  Image support:                                 Yes
  Max number of images read arguments:           128
  Max number of images write arguments:          64
  Max image 2D width:                            16384
  Max image 2D height:                           16384
  Max image 3D width:                            2048
  Max image 3D height:                           2048
  Max image 3D depth:                            2048
  Max samplers within kernel:                    16
  Max size of kernel argument:                   1024
  Alignment (bits) of base address:              2048
  Minimum alignment (bytes) for any datatype:    128
  Single precision floating point capability
    Denorms:                                     No
    Quiet NaNs:                                  Yes
    Round to nearest even:                       Yes
    Round to zero:                               Yes
    Round to +ve and infinity:                   Yes
    IEEE754-2008 fused multiply-add:             Yes
  Cache type:                                    Read/Write
  Cache line size:                               64
  Cache size:                                    16384
  Global memory size:                            16978542592
  Constant buffer size:                          4244635648
  Max number of constant args:                   8
  Local memory type:                             Scratchpad
  Local memory size:                             32768
  Max pipe arguments:                            16
  Max pipe active reservations:                  16
  Max pipe packet size:                          4244635648
  Max global variable size:                      3820172032
  Max global variable preferred total size:      16978542592
  Max read/write image args:                     64
  Max on device events:                          1024
  Queue on device max size:                      8388608
  Max on device queues:                          1
  Queue on device preferred size:                262144
  SVM capabilities:
    Coarse grain buffer:                         Yes
    Fine grain buffer:                           Yes
    Fine grain system:                           No
    Atomics:                                     No
  Preferred platform atomic alignment:           0
  Preferred global atomic alignment:             0
  Preferred local atomic alignment:              0
  Kernel Preferred work group size multiple:     64
  Error correction support:                      0
  Unified memory for Host and Device:            0
  Profiling timer resolution:                    1
  Device endianess:                              Little
  Available:                                     Yes
  Compiler available:                            Yes
  Execution capabilities:
    Execute OpenCL kernels:                      Yes
    Execute native function:                     No
  Queue on Host properties:
    Out-of-Order:                                No
    Profiling :                                  Yes
  Queue on Device properties:
    Out-of-Order:                                Yes
    Profiling :                                  Yes
  Platform ID:                                   00007FFA9DF66FD0
  Name:                                          gfx906
  Vendor:                                        Advanced Micro Devices, Inc.
  Device OpenCL C version:                       OpenCL C 2.0
  Driver version:                                2766.5 (PAL,HSAIL)
  Profile:                                       FULL_PROFILE
  Version:                                       OpenCL 2.0 AMD-APP (2766.5)
  Extensions:                                    cl_khr_fp64 cl_amd_fp64 cl_khr_global_int32_base_atomics cl_khr_global_int32_extended_atomics cl_khr_local_int32_base_atomics cl_khr_local_int32_extended_atomics cl_khr_int64_base_atomics cl_khr_int64_extended_atomics cl_khr_3d_image_writes cl_khr_byte_addressable_store cl_khr_fp16 cl_khr_gl_sharing cl_khr_gl_depth_images cl_amd_device_attribute_query cl_amd_vec3 cl_amd_printf cl_amd_media_ops cl_amd_media_ops2 cl_amd_popcnt cl_khr_d3d10_sharing cl_khr_d3d11_sharing cl_khr_dx9_media_sharing cl_khr_image2d_from_buffer cl_khr_spir cl_khr_subgroups cl_khr_gl_event cl_khr_depth_images cl_khr_mipmap_image cl_khr_mipmap_image_writes cl_amd_liquid_flash cl_amd_planar_yuv

**Comment by AndreaLanfranchi** [comment]
What is the complete output of clinfo ? Seems the initial part is missing.

**Comment by lesjokolat** [comment]
sorry missed it.. I just added it. I only copied the radeon vii  portion. would vega 64 and fury x clinfo help  ?

**Comment by gennadiv** [comment]
Yes the problem is on new drivers Adrenalin 2019 Edition 19.2.2 Optional why that miner ethminer release 0.18.0-alpha.3 sees not all video cards I have previously written about this with other versions of the miner works well

**Comment by AndreaLanfranchi** [comment]
Can you report `lspci` output ?

**Comment by lesjokolat** [comment]
HI im on windows 10. I tried googling the equivalent command but cant seem to find it. Do you know any way in windows or what command line would give you desired output?

**Comment by lesjokolat** [comment]
think I found what you needed

Listing OpenCL devices.
FORMAT: [platformID] [deviceID] deviceName
[0] [0] Fiji
        CL_DEVICE_TYPE: GPU
        CL_DEVICE_GLOBAL_MEM_SIZE: 4294967296
        CL_DEVICE_MAX_MEM_ALLOC_SIZE: 4026531840
        CL_DEVICE_MAX_WORK_GROUP_SIZE: 256
[0] [1] Fiji
        CL_DEVICE_TYPE: GPU
        CL_DEVICE_GLOBAL_MEM_SIZE: 4294967296
        CL_DEVICE_MAX_MEM_ALLOC_SIZE: 4026531840
        CL_DEVICE_MAX_WORK_GROUP_SIZE: 256
[0] [2] Fiji
        CL_DEVICE_TYPE: GPU
        CL_DEVICE_GLOBAL_MEM_SIZE: 4294967296
        CL_DEVICE_MAX_MEM_ALLOC_SIZE: 4026531840
        CL_DEVICE_MAX_WORK_GROUP_SIZE: 256
[0] [3] gfx906
        CL_DEVICE_TYPE: GPU
        CL_DEVICE_GLOBAL_MEM_SIZE: 16978542592
        CL_DEVICE_MAX_MEM_ALLOC_SIZE: 4244635648
        CL_DEVICE_MAX_WORK_GROUP_SIZE: 256
[0] [4] gfx900
        CL_DEVICE_TYPE: GPU
        CL_DEVICE_GLOBAL_MEM_SIZE: 8573157376
        CL_DEVICE_MAX_MEM_ALLOC_SIZE: 4244635648
        CL_DEVICE_MAX_WORK_GROUP_SIZE: 256

**Comment by JungleCatSW** [comment]
@lesjokolat You need to build ethminer using AMD APP sdk or maybe AMD OCL SDK, hunter is using some generic Open CL that doesn't recognise Open Cl from AMD driver v19+. I got it working using APP SDK.

**Comment by MaynardMiner** [comment]
Maybe I am doing it wrong- But I downloaded AMD APP from here:

https://github.com/GPUOpen-Tools/common-lib-amd-APPSDK-3.0

I then took the include and lib folder contents, and pasted them over the .hunter versions.

I then ran as usual. It didn't seem to work for me. I still get no opencl found.

**Comment by JungleCatSW** [comment]
@MaynardMiner Yeah it requires a bit more work in cmake, the find package has to be removed and some code modifications are required. I will setup a branch on my fork with the changes, and Ill post it here when it is done.

**Comment by MaynardMiner** [comment]
Damn...That's a lot of work to remove hunter. I started to do it, and was manually downloading all the dependencies, but I got stuck on linking the json packages (I'm not too familiar with them, not a lot of help out there for them, either).

It's weird though- I would think if I replaced the include dir and lib dir files with the one in .hunter folder, it would be more or less linking the updated libraries. Hunter doesn't seem to replace them after I did it.

**Comment by JungleCatSW** [comment]
@MaynardMiner you dont need to remove all of hunter, just the find_package(OpenCL),make it so that hunter manages all dependencies except for OpenCl. You could also make your own hunter package for and AMD OpenCL but it seemed like more work lol.

**Comment by lesjokolat** [comment]
I get this error when trying to build on windows

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:10.89
-- [hunter] Build step successful (dir: C:/.hunter/_Base/4b894e1/ee62265/192fd18/Build/CLI11)
-- [hunter] Cache saved: C:/.hunter/_Base/Cache/raw/7dc59afaca56e0c5d0dfb9e585157514b6c69bda.tar.bz2
CMake Warning (dev) at ethminer/CMakeLists.txt:17 (find_package):
  Policy CMP0074 is not set: find_package uses PackageName_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable CLI11_ROOT is set to:

    C:/.hunter/_Base/4b894e1/ee62265/192fd18/Install

  Environment variable CLI11_ROOT is set to:

    C:/.hunter/_Base/4b894e1/ee62265/192fd18/Install

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    API
    OPENCL


-- Build files have been written to: C:/Users/XXXX/Documents/GitHub/ethminer2/ethminer/build
Error: could not load cache
Press any key to continue . . .

any ideas appreciated

**Comment by MaynardMiner** [comment]
You are trying to build from wrong directory

**Comment by lesjokolat** [comment]
I tried this startup bat

@echo off
setlocal

rem add MSVC in PATH
call "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Community\Common7\Tools\VsMSBuildCmd.bat"

rem add Perl in PATH; it's needed for OpenSSL build
set "PERL_PATH=C:\Strawberry\perl\bin"
set "PATH=%PERL_PATH%;%PATH%"

rem switch to ethminer's source folder
cd "C:\Users\pablo\Documents\GitHub\ethminer2\ethminer"

if not exist "build\" mkdir "build\"

rem For CUDA 9.x pass also `-T v140`
set PATH="C:\Program Files\CMake\bin\";%PATH%
cmake -G "Visual Studio 15 2017 Win64" -H. -Bbuild -DOPENCL=ON -DETHASHCUDA=ON -DAPI=ON -DBINKERN=ON..
cmake --build . --config Release --target package

endlocal
pause

I have 2 versions of ethminer hence ethminer2 folder I build in.

**Comment by AndreaLanfranchi** [comment]
Duplicate #1840

**Comment by lesjokolat** [comment]
Thanks Maynard, fixed it to correct source. I am able to build but still get 

ethminer 0.18.0-rc.0
Build: windows/release/msvc

No OpenCL platforms found
Error: No usable mining devices found

**Comment by MaynardMiner** [comment]
The problem is that hunter gets an older version of openCL sdk than the new drivers (18.16.1) require.

For some reason, and I'm not sure why- But if you install an older version of the drivers (18.6.1 worked for me), then upgrade to recent- It retains whatever openCL information is required to get ethminer to work.

Eventually hunter repo will be updated, and will work- But for now install 18.6.1 drivers, then install latest drivers OVER them (don't do clean install). That should fix it.

**Comment by rileyweber13** [comment]
Does anyone have links to install an older, working driver on linux?

**Comment by lesjokolat** [comment]
at bottom of latest drivers ther is a small link previous drivers goes some back...


https://www.amd.com/en/support/previous-drivers/graphics/amd-radeon-2nd-generation-vega/amd-radeon-2nd-generation-vega/amd-radeon-vii

BTW new 19.30 maybe try that one...

**Comment by doksec** [comment]
win10 RX580 drivers 19.9.3

`bin\ethminer.exe"` -G --list-devices
ethminer 0.18.0
Build: windows/release/msvc

No OpenCL platforms found
Error: No usable mining devices found`

**Comment by slinkyvagabond** [comment]
Do you have the OCL_SDK_Light installed?

**Comment by doksec** [comment]
OCL_SDK_Light install - don't work!
```
ethminer 0.18.0
Build: windows/release/msvc

No OpenCL platforms found
Error: No usable mining devices found
```
```
ethminer 0.19.0-alpha.0
Build: windows/release/msvc

No OpenCL platforms found
Error: No usable mining devices found
```

**Comment by jam3st** [comment]
Same problem.  Ran procmon on clinfo and noticed that it was picking up the OpenCL libs from C:\Windows\System32\DriverStore\FileRepository\u0347413.inf_amd64_09c106fe9d27c4e3\B347422\amdocl64.dll
I was also reading the reg key HLKM\software\khronos\OpenCL\vendors, so adding a DWORD key wit that name "C:\Windows\System32\DriverStore\FileRepository\u0347413.inf_amd64_09c106fe9d27c4e3\B347422\amdocl64.dll" set to zero means the device is found.

You could probably also copy that file, but this is just so stuffed up.

**Comment by jgonzi** [comment]
> 
> 
> Same problem. Ran procmon on clinfo and noticed that it was picking up the OpenCL libs from C:\Windows\System32\DriverStore\FileRepository\u0347413.inf_amd64_09c106fe9d27c4e3\B347422\amdocl64.dll
> I was also reading the reg key HLKM\software\khronos\OpenCL\vendors, so adding a DWORD key wit that name "C:\Windows\System32\DriverStore\FileRepository\u0347413.inf_amd64_09c106fe9d27c4e3\B347422\amdocl64.dll" set to zero means the device is found.
> 
> You could probably also copy that file, but this is just so stuffed up.

The perfect solution... It works great!

**Comment by ONLY-VEDA** [comment]
> Same problem. Ran procmon on clinfo and noticed that it was picking up the OpenCL libs from C:\Windows\System32\DriverStore\FileRepository\u0347413.inf_amd64_09c106fe9d27c4e3\B347422\amdocl64.dll
> I was also reading the reg key HLKM\software\khronos\OpenCL\vendors, so adding a DWORD key wit that name "C:\Windows\System32\DriverStore\FileRepository\u0347413.inf_amd64_09c106fe9d27c4e3\B347422\amdocl64.dll" set to zero means the device is found.
> 
> You could probably also copy that file, but this is just so stuffed up.

thanks so much

**Comment by mattmarano** [comment]
Solution here: https://github.com/ethereum-mining/ethminer/issues/2001

**Comment by mansoor96g** [comment]
i'm having the same problem but i'm running linux
